packages/api/src/platforms/vtex/resolvers/query.ts

```diff
@@ -11,22 +11,26 @@ import type {
 } from '../../../__generated__/schema'
 import type { CategoryTree } from '../clients/commerce/types/CategoryTree'
 import type { Context } from '../index'
+import ChannelMarshal from '../utils/channel'
 
 export const Query = {
   product: async (_: unknown, { locator }: QueryProductArgs, ctx: Context) => {
     // Insert channel in context for later usage
     ctx.storage = {
       ...ctx.storage,
       channel:
-        locator.find((facet) => facet.key === 'channel')?.value ??
-        ctx.storage.channel,
+        // TODO: Check if make sense move this parse a layer down or pass the whole parsed channel
```

Comment:
just keep it like a string

Reply:
Keep like a string breaks some internal field resolvers take a look at the 
https://sfj-a1d5edd--base.preview.vtex.app/office/

```
 ERROR

Error: Product with id 2791588 has no sellers for channel
{"regionId":"v2.1BB18CE648B5111D0933734ED83EC783","salesChannel":"1"}.
    at Object.offers
(webpack://base.store/./node_modules/@faststore/api/dist/api.esm.js?:820:13)
    at Object.field.resolve [as StoreProductoffersResolver]
(webpack://base.store/./node_modules/@envelop/core/index.js?:86:32)
    at nodeStoreProductEdgenodeResolverHandler (eval at compileQuery
(webpack://base.store/./node_modules/graphql-jit/dist/execution.js?), <anonymous>:3012:39)
    at StoreProductConnection__fieldedgesMapItemHandler.__value.then.__context.nullErrors.push._
_context.GraphQLError.line (eval at compileQuery
(webpack://base.store/./node_modules/graphql-jit/dist/execution.js?), <anonymous>:3151:9)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  locations: [ { line: 32, column: 11 } ],
  path: [ 'search', 'products', 'edges', 0, 'node', 'offers' ]
}
```